services/headless-lms/server/src/controllers/main_frontend/organizations.rs

```diff
@@ -38,13 +38,30 @@ GET `/api/v0/main-frontend/organizations/{organization_id}/courses"` - Returns a
 async fn get_organization_courses(
     organization_id: web::Path<Uuid>,
     pool: web::Data<PgPool>,
+    user: Option<AuthUser>,
     pagination: web::Query<Pagination>,
 ) -> ControllerResult<web::Json<Vec<Course>>> {
     let mut conn = pool.acquire().await?;
-    let courses =
-        models::courses::organization_courses_paginated(&mut conn, &organization_id, &pagination)
-            .await?;
-    Ok(web::Json(courses))
+    let courses = models::courses::organization_courses(&mut conn, *organization_id).await?;
+    let mut viewable_courses = vec![];
+    for course in courses {
+        // only add draft courses that the user is allowed to view
+        // todo: think of a more efficient method
```

Comment:
Instead of authorizing stuff in a loop, the best way is to implement a function that does the filtering in SQL. The function could be something like this:

```rust
async fn organization_courses_visible_to_user_paginated(conn: x, organization_id: Uuid, user_id: Uuid) {
  if "user not logged in" {
    return "non-draft courses paginated"
  }
  if "the user has a view permission on the organization or globally" {
    return "all courses in organization paginated"
  }
  # I think we can assume for now that all roles have the view ability 
  return "non-draft courses and courses where the user has some role"
}
```